src/logic/dijkstra.ts

```diff
@@ -1,68 +1,89 @@
 import { Graph } from './commonInterfaces';
 
-interface Nodes {
-    [key: string]: number
+interface Costs {
+    [key: number]: number
 }
 
-interface PathsMap {
-    [key: string]: number[]
+interface Paths {
+    [key: number]: Array<number>
 }
 
-
-export class Result {
-    nodes: Nodes;
-    pathMap: PathsMap;
-
-    constructor(distances: Nodes, pathMap: PathsMap) {
-      this.nodes = distances;
-      this.pathMap = pathMap;
-    }
+interface TraversalRecord{
+  node: number,
+  lookups: Array<number>
 }
 
-export default function calculatePaths(graph: Graph, startNode: number = 0) {
-  // TODO: doccomment
-  const nodes = Array.from(Array(graph.length).keys());
-  const pathsMap: PathsMap = nodes.reduce(
-    (acc: PathsMap, curr: number) => (acc[curr] = [], acc), {}
-  );
-  const unvisitedNodes: Set<number> = new Set(nodes);
-  const costs: Nodes = nodes.reduce((acc: Nodes, curr: number) =>
-    (acc[curr] = Infinity, acc), {});
-
-  costs[startNode] = 0;
-  let currentNode = startNode;
+export class dijkstraTracer{
+    costs: Costs;
+    paths: Paths;
+    traversalHistory: Array<TraversalRecord>;
+    #graph: Graph;
+    #startNode: number;
+    
+    constructor(graph: Graph, startNode: number = 0) {
+      this.#graph = graph;
+      this.#startNode = startNode;
+      this.traversalHistory = [];
+      this.costs = {};
+      this.paths = {};
+      this.setCostsPathsAndTraversalHistory();
+    }
 
-  while (unvisitedNodes.size) {
-    for (const [neighboredNode, neighbourCost] of graph[currentNode].entries()) {
-      if (neighbourCost !== 0) {
-        const previousCost = costs[neighboredNode];
-        const currentCost = neighbourCost + costs[currentNode];
+    setCostsPathsAndTraversalHistory() {
```

Comment:
Am I suppose to run this from outside? Either this also gets `_` or `_getTravelsalFor` shouldn't too.

Reply:
_getTravelsalFor maybe actually be useful from outside in some cases.

Reply:
Good point, this method (`setCostsPathsAndTraversalHistory`) should be private